### PR TITLE
[installer]: validate that cluster meets minimum kubernetes version

### LINKED ISSUE #6828: [Installer]: establish minimum K8s version and set validation check

### PR DESCRIPTION
**Pull request by MrSimonEmms**
## Description

Validate that the min Kubernetes version should be `1.21.0-0` - using pre-release to allows for providers adding their own `patch` to the version

## Related Issue(s)

Fixes #6828

## How to test

```shell
gitpod-installer validate cluster --kubeconfig /path/to/kubeconfig
```

## Release Notes

```release-note
Validate cluster min k8s version
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **mrsimonemms** after the PR has been reviewed.

Associated issue: *#6828*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: installer/pkg/cluster/checks.go [hunk 1]
```diff
@@ -99,6 +106,45 @@ func checkContainerDRuntime(ctx context.Context, config *rest.Config, namespace
...
+
+	// Allow pre-releases in case provider appends anything to the version
+	constraint, err := semver.NewConstraint(kubeletVersionConstraint)
+	if err != nil {
+		return nil, err
+	}
+
+	var res []ValidationError
+	for _, node := range nodes {
+		kubeletVersion := node.Status.NodeInfo.KubeletVersion
```
Comment (csweichel) [inline installer/pkg/cluster/checks.go:123]: We should also ensure the version of the API/control plane matches what we expect (I'd actually think that more important than the kubelet version).

`kubectl version` offers an example on how to do that: https://github.com/kubernetes/kubectl/blob/94d80be6aba709add4dc20db166fa8c54d176653/pkg/cmd/version/version.go#L126

**Comment by MrSimonEmms** [comment]
@csweichel I've applied the check on the Kubernetes version as requested. I've also kept a check on a per-node basis. As I see it, we have the following options:
1. use `KubeletVersion`
2. use `KubeProxyVersion`
3. do nothing

I must confess, I don't know the difference between 1 and 2 but it feels like 3 might cause problems - it's possible to have the control pane on one version, but the individual nodes on a different (lower) version.

I'm open to suggestions

**Comment by csweichel** [comment]
I reckon we could merge this change as is and see how the version check fairs in the wild.
There's a minor nit, but feel free to unhold and fix in a later PR.

/hold
/approve
/lgtm

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: ee85a296cf4c5bc8a7b613c3aaac80eac23cb642</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6926#issuecomment-980541264" title="LGTM">csweichel</a>*

Associated issue: *#6828*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 27550a8e92a99f63a15667210bffa117391a94ee</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6926#issuecomment-981579931" title="LGTM">csweichel</a>*

Associated issue: *#6828*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**